### ISSUE TITLE
NR error: undefined method `each' for nil:NilClass in DataMagic.search

### ISSUE THREAD

**Issue by yozlet** [18F/open-data-maker#276]
[New Relic error log](https://rpm.newrelic.com/accounts/1065766/applications/7790612/filterable_errors#/show/c5b584-592d21ab-c3b2-11e5-bb22-b82a72d2466d/stack_trace?top_facet=transactionUiName&primary_facet=error.class&_k=n4gixl). Error triggered by loading this URL on staging: https://ccapi-staging.18f.gov/v1/schools/stats?school.degrees_awarded.predominant=2,3&_fields=2013.student.size&_metrics=avg,std_deviation,std_deviation_bounds

I _think_ it's because the `school.degrees_awarded.predominant` field doesn't exist - so why isn't ErrorChecker trapping it?

Traceback:
```
NoMethodError: undefined method `each' for nil:NilClass
                       /home/vcap/app/lib/data_magic.rb:  142:in `search'
                      /home/vcap/app/app/controllers.rb:   69:in `search_and_respond'
                      /home/vcap/app/app/controllers.rb:   65:in `process_params'
                      /home/vcap/app/app/controllers.rb:   51:in `block (2 levels) in <top (required)>'
```

**Comment by yozlet** [comment]
Oops, my mistake. The problem with the above query is the `_fields=2013.student.size` part. It appears that we get this error with any year-nested fields. Aggregating on `schools.*` fields works fine.

I think the problem is happening in the query formulation. Here's the query generated for aggregating on `school.tuition_revenue_per_fte`:
```ruby
{:index=>"development-school-data", :type=>"document", :body=>{:from=>0, :size=>20, :query=>{:match_all=>{}}, :aggs=>{"school.tuition_revenue_per_fte"=>{:extended_stats=>{"field"=>"school.tuition_revenue_per_fte"}}}, :fields=>["school.tuition_revenue_per_fte"], :_source=>false}, :search_type=>"count"}
```

and here's the query for aggregating on `2013.cost.avg_net_price.overall`:
```ruby
{:index=>"development-school-data", :type=>"document", :body=>{:from=>0, :size=>20, :query=>{:match_all=>{}}, :fields=>["2013.cost.avg_net_price.overall"], :_source=>false}, :search_type=>"count"}
```

The latter produces no `aggregations` element in the returned JSON, which triggers the above error. But when I formulate the query the right way in Sense (with the added `extended_stats` sub-hash), it works fine.

@siruguri, any ideas? I can dive into this but it may take me a little while to get to.

**Comment by siruguri** [comment]
which branch has the code that I can check this on? master?

**Comment by yozlet** [comment]
Yep, this is in master now (as of last week) but otherwise try `dev`.

**Comment by yozlet** [comment]
Also, this bug only surfaces with nested data - look at the `spec/fixtures/nested-files` and `spec/fixtures/nested2` fixtures.

**Comment by siruguri** [comment]
I think I'll have time early next week to look into this ... lmk if you get
to it earlier. I'm guessing it's something to do with the check for types
in `lib/data_magic/query_builder.rb` line 45. Maybe the nested fields cause
the `column_field_type` to not be set to integer or float?

**Comment by ultrasaurus** [comment]
actually, I think in this case it is because the [resulting values](http://localhost:3000/v1/schools?school.degrees_awarded.predominant=2,3&_fields=2013.student.size) are all `null` so I guess the stats aren't calculated and `result['aggregations']` is nil

**Comment by yozlet** [comment]
Reopening because this issue isn't fixed, though thanks to #286 it's no longer throwing exceptions; it just returns empty results. Still needs a failing test.
